core/src/main/java/org/elasticsearch/action/support/AutoCreateIndex.java

```diff
@@ -67,11 +69,13 @@ public boolean needToCheck() {
     /**
      * Should the index be auto created?
      */
-    public boolean shouldAutoCreate(String index, ClusterState state) {
+    public boolean shouldAutoCreate(String index, ClusterState state, IndexNameExpressionResolver resolver) {
         if (!needToCheck) {
             return false;
         }
-        if (state.metaData().hasConcreteIndex(index)) {
+        IndicesOptions options = IndicesOptions.fromOptions(true, true, true, true);
+        String[] concreteIndices = resolver.concreteIndices(state, options, index);
+        if (concreteIndices.length > 0) {
```

Comment:
yes, the date math needs to be expanded... we can add a concreteSingleIndex() variant that doesn't need indices options?